frame/staking/src/lib.rs

```diff
@@ -1391,6 +1548,11 @@ decl_module! {
 			add_weight(3, 0, 0);
 			// Additional read from `on_finalize`
 			add_weight(1, 0, 0);
+
+			let task_weight = <SlashTaskExecutor<T>>::mutate(|e| e.execute());
+			// The additional weight of reading the tasks, and writing back the result.
+			add_weight(1, 1, task_weight);
```

Comment:
This does not only add the storage accesses but also decoding and encoding weight proportional to the amount of queued tasks.

Reply:
Indeed, this is one of the things that need to be fixed. Not sure where, but there's a TODO for it.